docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
+the eventing mesh. If an event type is not ready for consumption, we should explicitly indicate so.
+3. The event types stored in the Registry should contain all the required information 
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+## Design Ideas
+
+### EventType CRD
+
+We propose having a namespaced EventType CRD. Here is an example of how a CR would look like:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repopush
+spec:
+  type: repo:push
+  source: my-user/my-repo
+  schema: my-schema
+  broker: default
+```
+
+
+- The `name` of the EventType is advisory, non-authoritative. Given that Cloud Event types can 
+contain characters that may not comply with Kubernetes naming conventions, we will (slightly) 
+modify those names to make them K8s-compliant, whenever we need to generate them. 
+
+- `type` is authoritative. This refers to the Cloud Event type as it enters into the eventing mesh. 
+
+- `source`: an identifier of where we receive the event from. This might not necessarily be the Cloud Event source 
+attribute. If we receive the event from our receive adaptors, the info might come in a Cloud Event custom extension (e.g., from).
+
+- `schema` is a URI with the EventType schema. It may be a JSON schema, a protobuf schema, etc. It is optional.
+
+- `broker` refers to the Broker that can provide the EventType. 
```

Comment:
With the current proposal, yes. You will expect to have two EventType CRs. But it might not be correct though. 
I think having it as a string it makes it easier for checking in the Broker ingress whether the EventType is registered or not using a label selector on the broker name.

Reply:
I used `broker` as the TriggerSpec uses `broker` instead of `brokerName`, but I'm ok with both.

I'm not entirely sure if we should add the name of the broker as part of the EventType name. Maybe in the future we want to  support cross-broker EventTypes, and so on, and the names might be confusing? Though not sure.

I was thinking that by adding additional columns (e.g.., broker) to the output of kubectl get eventtypes, we could easily identify the EventTypes of each broker.